tools/init.sh

```diff
@@ -108,6 +114,23 @@ case "$1" in
            echo "PEER_ID: $PEER"
            echo "TCE_LOCAL_KS: $HOSTNAME"
 
+           INDEX=0
+
+           # For libp2p_keys.json
+           LIBP2P_KEY=$(jq -r ".keys[$INDEX]" /tmp/shared/libp2p_keys.json)
+           echo -n "$LIBP2P_KEY" > $TOPOS_HOME/node/test/libp2p/libp2p.key
+           jq "del(.keys[$INDEX])" /tmp/shared/libp2p_keys.json > /tmp/shared/libp2p_keys_temp.json && mv /tmp/shared/libp2p_keys_temp.json /tmp/shared/libp2p_keys.json
```

Comment:
If the nodes are spawn at the same time, they'll all have the `INDEX=0` key right?

maybe we can use a random index like:

```
INDEX=$(jq --arg i $(($RANDOM % $(jq length  /tmp/shared/libp2p_keys.json))) '$i|tonumber'  /tmp/shared/libp2p_keys.json)
```

Reply:
Since this is blocking, we can always have the same `index`. We lock the file and access it, write it back, release the lock. So every peer just always takes the first from the array without a problem.

Reply:
Which part are you defining as blocking?
Reading a file doesn't create a lock and writing to it either because it's a shared file, that's why we end up having lock mechanisms for the creation of the peers.

Reply:
Ups, exactly. I had a `lock`  version locally as an experiment and somehow a different version made it upstream. My bad. Yeah, either using a lock like with the `peer_list` or using a random index 👍